src/parser/SparqlParserHelpers.cpp

```diff
@@ -93,12 +93,19 @@ ResultOfParseAndRemainingText<vector<GroupKey>> parseGroupClause(
 }
 // _____________________________________________________________________________
 
-ResultOfParseAndRemainingText<std::optional<GraphPatternOperation::Values>>
-parseValuesClause(const std::string& input,
-                  SparqlQleverVisitor::PrefixMap prefixes) {
+ResultOfParseAndRemainingText<GraphPatternOperation::Values> parseDataBlock(
+    const std::string& input, SparqlQleverVisitor::PrefixMap prefixes) {
+  ParserAndVisitor p{input, std::move(prefixes)};
+  return p.parseTypesafe(input, "data block",
+                         &SparqlAutomaticParser::dataBlock);
+}
```

Comment:
Since this is again a small PR we can improve our abstractions.
I think with the improved `parseTypesafe` abstraction we can improve this function to fully generically become
`parseFront<&SparqlAutomaticParser::dataBlock, "data block">(input, prefixes)`.

(The value type of the `ResultOfParseAnd...` can now be automatically deduced with the typesafe parsing).

First try it yourself, and let me know if I can be of service.